File: 0094.binary_tree_inorder_traversal/solution.py
from typing import List

import utils


class Solution:
    def inorder_traversal(self, root: utils.TreeNode) -> List[int]:
        def inorder(node, ans):
            if not node:
                return ans
            ans = inorder(node.left, ans)
            ans.append(node.val)
            ans = inorder(node.right, ans)
            return ans
        return inorder(root, [])

    def loop(self, root):
        stack, ans = [], []
        while root or stack:
            while root:
                stack.append(root)
                root = root.left
            root = stack.pop()
            ans.append(root.val)
            root = root.right
        return ans

File: 0094.binary_tree_inorder_traversal/solution.go
package leetcode0094

import "github.com/Ackerr/Algorithms/utils"

func inorderTraversal(root *utils.TreeNode) []int {
	if root == nil{
		return []int{}
	}
	var stack []*utils.TreeNode
	var ans []int
	for root != nil || len(stack) > 0{
		for root != nil{
			stack = append(stack, root)
			root = root.Left
		}
		index := len(stack) - 1
		root = stack[index]
		ans = append(ans, root.Val)
		root = root.Right
		stack = stack[:index]
	}
	return ans
}